src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -148,10 +148,14 @@ public void visitToken(DetailAST ast) {
             leftCurly = slistToken;
         }
 
-        if (leftCurly != null) {
+        if (leftCurly != null || ast.getType() == TokenTypes.LITERAL_CASE) {
```

Comment:
As soon leftCurly is found and calculated correctly. 
Check should be not differ validation of block. 
All blocks are identical `{   }`, the difference is only to what parent they belong to.

If I miss smth , please give me there links to input cases that will not follow that rule and really require so special processing.